File: src/main/java/com/sstep/recipe/manager/vo/IngredientVo.java
package com.sstep.recipe.manager.vo;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.sstep.recipe.manager.enums.UnitOfMeasure;


public class IngredientVo {
    private String name;
    private UnitOfMeasure unitOfMeasure;
    private Double quantity;

    @JsonCreator
    public IngredientVo(
            @JsonProperty("name") final String name,
            @JsonProperty("unitOfMeasure") final UnitOfMeasure unitOfMeasure,
            @JsonProperty("quantity") final Double quantity) {
        this.name = name;
        this.unitOfMeasure = unitOfMeasure;
        this.quantity = quantity;
    }

    public String getName() {
        return name;
    }

    public void setName(final String name) {
        this.name = name;
    }

    public UnitOfMeasure getUnitOfMeasure() {
        return unitOfMeasure;
    }

    public void setUnitOfMeasure(final UnitOfMeasure unitOfMeasure) {
        this.unitOfMeasure = unitOfMeasure;
    }

    public Double getQuantity() {
        return quantity;
    }

    public void setQuantity(final Double quantity) {
        this.quantity = quantity;
    }
}

File: src/main/kotlin/com/sstep/recipe/manager/vo/IngredientVo.kt
package com.sstep.recipe.manager.vo

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty
import com.sstep.recipe.manager.enums.UnitOfMeasure

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
data class IngredientVo @JsonCreator constructor(
    val name: String? = null,
    val unitOfMeasure: UnitOfMeasure? = null,
    val quantity: Double? = null
)